riptide-core/src/main/java/org/zalando/riptide/Route.java

```diff
@@ -22,28 +31,30 @@
 
 import com.google.common.reflect.TypeParameter;
 import com.google.common.reflect.TypeToken;
-import org.springframework.http.HttpHeaders;
-import org.springframework.http.HttpMessage;
-import org.springframework.http.ResponseEntity;
-import org.springframework.http.client.ClientHttpResponse;
-
-import java.io.IOException;
-import java.net.URI;
-import java.util.List;
 
 @FunctionalInterface
 public interface Route {
 
     void execute(final ClientHttpResponse response, final MessageReader reader) throws Exception;
 
     static Route call(final ThrowingRunnable consumer) {
-        return (response, reader) ->
+        return (response, reader) -> {
+            try {
                 consumer.run();
+            } finally {
+                response.close();
+            }
+        };
     }
 
     static Route call(final ThrowingConsumer<ClientHttpResponse> consumer) {
-        return (response, reader) ->
+        return (response, reader) -> {
+            try {
                 consumer.accept(response);
+            } finally {
+                response.close();
```

Comment:
This would be in the anonymous dispatcher in the Requester.
```
future.addCallback(response -> {
                    try { ...
```
Doing it here would close the stream in any case. I think this would be okay for all use cases. Anyhow, I can't see how this would result in suppressed exceptions without having code coverage problems.